### PR TITLE
gov: pools gov prop supercharge liquidity

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #5307 

## What is the purpose of the change

We currently only allow for a single pool to be added at a time via gov prop. This PR allows for multiple pools to be created with a single gov message.

## Testing and Verifying

Tests modified to expect a list of poolRecords rather than just one

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: proto/osmosis/concentrated-liquidity/gov.proto [hunk 1]
```diff
@@ -5,30 +5,18 @@ import "gogoproto/gogo.proto";
...
-    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
-    (gogoproto.moretags) = "yaml:\"exponent_at_price_one\"",
-    (gogoproto.nullable) = false
-  ];
-  string spread_factor = 7 [
-    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
-    (gogoproto.moretags) = "yaml:\"spread_factor\"",
-    (gogoproto.nullable) = false
-  ];
+  repeated PoolRecord pool_records = 3 [ (gogoproto.nullable) = false ];
```
Comment (AlpinYukseloglu) [inline proto/osmosis/concentrated-liquidity/gov.proto:19]: Should we be adding a tag here? e.g  `(gogoproto.moretags) = "yaml:\"pool_records\""`

File: x/concentrated-liquidity/types/gov.go [hunk 2]
```diff
@@ -15,71 +15,84 @@ const (
...
+func (p CreateConcentratedLiquidityPoolsProposal) String() string {
+	recordsStr := ""
+	for _, record := range p.PoolRecords {
+		recordsStr = recordsStr + fmt.Sprintf("(Denom0: %s, Denom1: %s, TickSpacing: %d, ExponentAtPriceOne: %d, SpreadFactor: %d) ", record.Denom0, record.Denom1, record.TickSpacing, record.ExponentAtPriceOne, record.SpreadFactor)
 	}
 
-	return nil
+	var b strings.Builder
+	b.WriteString(fmt.Sprintf(`Create Concentrated Liquidity Pool Proposal:
+  Title:                 %s
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/types/gov.go:91]: nit: this formatting looks off

File: x/concentrated-liquidity/types/gov_test.go [hunk 3]
```diff
@@ -10,39 +10,52 @@ import (
 	"github.com/osmosis-labs/osmosis/v15/x/concentrated-liquidity/types"
 )
 
-func TestCreateConcentratedLiquidityPoolProposalMarshalUnmarshal(t *testing.T) {
+func TestCreateConcentratedLiquidityPoolsProposalMarshalUnmarshal(t *testing.T) {
+	records := []types.PoolRecord{
+		{
+			Denom0:             "uion",
+			Denom1:             "uosmo",
+			TickSpacing:        100,
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/types/gov_test.go:18]: nit: we should have different tick spacings here as well & if possible another set of records (non-blocking)

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM, nice work. Left a couple nits but approving since none are blocking. Note that I only reviewed diff & didn't test new cli command on localosmosis

**Comment by czarcas7ic** [comment]
@AlpinYukseloglu addressed all comments in [this commit](https://github.com/osmosis-labs/osmosis/pull/5345/commits/7a15c33e8f7d5da972bb9b101756dbdf8f6bc1d2), thanks!

**Pull request merged**